python/phy_layer.py

```diff
@@ -449,13 +483,19 @@ def initialize_usrps(
 
         self.logger.debug(f"{usrp_device_args=}")
 
+        devices = None
+        if "serial" in rx_device_addr or "serial" in tx_device_addr:
+            self.logger.info("Searching device infos to check the types.")
+            devices = tacmac.uhd_configuration.find_devices()
```

Comment:
This should always (or in most cases) be true. Thus, the `find_devices` part would be executed again.